libraries/AP_RangeFinder/AP_RangeFinder_analog.cpp

```diff
@@ -51,10 +51,7 @@ AP_RangeFinder_analog::AP_RangeFinder_analog(RangeFinder::RangeFinder_State &_st
 */
 bool AP_RangeFinder_analog::detect(AP_RangeFinder_Params &_params)
 {
-    if (_params.pin != -1) {
-        return true;
```

Comment:
Problem with some of these is that if we want to add more checks then we'd probably restructure to look like it currently does.  Otherwise people start to stack the logic up on one line and that gets nasty.  So the *pattern* is important in some of these.

Reply:
Done.